python/lsst/cp/pipe/utils.py

```diff
@@ -540,6 +540,37 @@ def arrangeFlatsByExpTime(exposureList):
     return flatsAtExpTime
 
 
+def arrangeFlatsByExpId(exposureList):
+    """Arrange exposures by exposure ID.
+
+    Parameters
+    ----------
+    exposureList : `list`[`lsst.afw.image.exposure.exposure.ExposureF`]
+        Input list of exposures.
+
+    Returns
+    ------
+    flatsAtExpTime : `dict` [`float`,
+                      `list`[`lsst.afw.image.exposure.exposure.ExposureF`]]
+        Dictionary that groups flat-field exposures that have the same
+        exposure time (seconds).
+    """
+    flatsAtExpId = {}
+    sortedExposures = sorted(exposureList, key=lambda exp: exp.getInfo().getVisitInfo().getExposureId())
+
+    for jPair, exp in enumerate(sortedExposures):
+        if (jPair + 1) % 2:
+            kPair = jPair // 2
+            listAtExpId = flatsAtExpId.setdefault(kPair, [])
+            listAtExpId.append(exp)
+            try:
+                listAtExpId.append(sortedExposures[jPair + 1])
```

Comment:
Is this line here for when the number of exposures is not even?  If not, what happens when this is the case? or when there are more than two exposures with the same flux? What if there are 3 exposures with the same flux, and then another one with a different flux, so in the end you end up with a pair at the same flux (the first two), but then a pair with fluxes that don't match (exposures 3 and 4 in my example)?  Maybe you ready took this into account.

Reply:
I've moved the first append into the try block.  This will resolve the case of an odd number of input exposures (as the second append will trigger the rollback of the first exposure as well, preventing partially populated pairs).  The case of bad inputs is not handled.  My solution here was mostly a way to make the best of the imSim situation, as I didn't think that simply sorting by mean would work.  Pathological cases like `flux=[1000, 1001, 1002, 1010]` where there might be better choices convinced me that using the simplest solution would be best.